google/cloud/security/common/gcp_api/cloud_logging.py

```diff
@@ -0,0 +1,443 @@
+"""
+
+import json
+import logging
+
+from google.cloud.security.common.gcp_api._base_client import _BaseClient
+# pylint: disable=line-too-long
+from google.cloud.security.common.third_party.transports.background_thread import BackgroundThreadTransport
+from google.protobuf.json_format import MessageToJson
+
+
+_RFC3339_MICROS = '%Y-%m-%dT%H:%M:%S.%fZ'
+
+
+def _datetime_to_rfc3339(value, ignore_zone=True):
+    """Convert a timestamp to a string.
+
+    Args:
+        :type value: :class:`datetime.datetime`
+        :param value: The datetime object to be converted to a string.
+
+        :type ignore_zone: bool
+        :param ignore_zone: If True, then the timezone (if any) of the datetime
+                            object is ignored.
+
+        :rtype: str
+        :returns: The string representing the datetime stamp.
+    """
+    if not ignore_zone and value.tzinfo is not None:
+        # Convert to UTC and remove the time zone info.
+        value = value.replace(tzinfo=None) - value.utcoffset()
+
+    return value.strftime(_RFC3339_MICROS)
+
+
+class LoggingClient(_BaseClient):
+    """Logging Client."""
+
+    API_NAME = 'logging'
+
+    def __init__(self, credentials=None):
+        super(LoggingClient, self).__init__(
+            credentials=credentials, api_name=self.API_NAME)
+
+    def get_logger(self, name=__name__):
+        """Get the Logger."""
+        return CloudLogger(name, self)
+
+    def write_entries(self, entries, logger_name=None, resource=None,
+                      labels=None):
+        """API call:  log an entry resource via a POST request
+
+        See:
+        https://cloud.google.com/logging/docs/api/reference/rest/v2/entries/write
+
+        Args:
+            :type entries: sequence of mapping
+            :param entries: the log entry resources to log.
+
+            :type logger_name: str
+            :param logger_name: name of default logger to which to log the
+                                entries; individual entries may override.
+
+            :type resource: mapping
+            :param resource: default resource to associate with entries;
+                             individual entries may override.
+
+            :type labels: mapping
+            :param labels: default labels to associate with entries;
+                           individual entries may override.
+        """
+        data = {'entries': list(entries)}
+
+        if logger_name is not None:
+            data['logName'] = logger_name
+
+        if resource is not None:
+            data['resource'] = resource
+
+        if labels is not None:
+            data['labels'] = labels
+
+        self.service.entries().write(body=data).execute()
+
+
+class CloudLoggingHandler(logging.StreamHandler):
+    # pylint: disable=line-too-long
+    """Cloud Logging Handler.
+
+    See: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/logging/google/cloud/logging/handlers/handlers.py
```

Comment:
Is there an issue with just directly using google-cloud-python instead of importing pieces of it? How do we get patches / updates if there are bug fixes?